src/ContentClient.ts

```diff
@@ -36,25 +37,37 @@ import {
   splitAndFetch,
   splitValuesIntoManyQueryBuilders
 } from './utils/Helper'
-import { DeploymentData } from './utils/DeploymentBuilder'
+import { DeploymentBuilder, DeploymentData, DeploymentPreparationData } from './utils/DeploymentBuilder'
 import NodeFormData from 'form-data'
 
 export class ContentClient implements ContentAPI {
   private static readonly CHARS_LEFT_FOR_OFFSET = 7
   private readonly contentUrl: string
   private readonly fetcher: Fetcher
+  private readonly deploymentBuilderClass: typeof DeploymentBuilder
 
   constructor(
     contentUrl: string,
     private readonly origin: string, // The name or a description of the app that is using the client
-    fetcher?: Fetcher
+    fetcher?: Fetcher,
+    deploymentBuilderClass?: typeof DeploymentBuilder
   ) {
     this.contentUrl = sanitizeUrl(contentUrl)
     this.fetcher =
       fetcher ??
       new Fetcher({
         headers: getHeadersWithUserAgent('content-client')
       })
+    this.deploymentBuilderClass = deploymentBuilderClass ?? DeploymentBuilder
+  }
+
+  async buildDeployment(type: EntityType,
```

Comment:
I'd name it as `buildEntity` as you are creating the entity, the deployment exists only when you deployed it.

Reply:
Will do